idb/postgres/postgres_integration_test.go

```diff
@@ -668,4 +669,247 @@ func TestIgnoreDefaultFrozenConfigUpdate(t *testing.T) {
+		assert.Equal(t, sql.NullBool{Valid: true, Bool: true}, deleted)
+		assert.Equal(t, sql.NullInt64{Valid: true, Int64: int64(test.Round)}, createdAt)
+		assert.Equal(t, sql.NullInt64{Valid: true, Int64: int64(test.Round + 1)}, closedAt)
+	}
+}
+
+func TestDestroyAssetZeroSupply(t *testing.T) {
+	db, connStr, shutdownFunc := setupPostgres(t)
+	defer shutdownFunc()
+
+	pdb, err := idb.IndexerDbByName("postgres", connStr, nil, nil)
+	assert.NoError(t, err)
+
+	cache, err := pdb.GetDefaultFrozen()
+	assert.NoError(t, err)
+
+	assetID := uint64(3)
+
+	state := getAccounting(test.Round, cache)
+
+	// Create an asset.
+	{
+		// Set total supply to 0.
+		_, txnRow := test.MakeAssetConfigOrPanic(test.Round, 0, assetID, 0, 0, false, "uu", "aa", "",
+			test.AccountA)
+
+		err := state.AddTransaction(txnRow)
+		assert.NoError(t, err)
+	}
+	// Destroy an asset.
+	{
+		_, txnRow := test.MakeAssetDestroyTxn(test.Round, assetID)
+
+		err := state.AddTransaction(txnRow)
+		assert.NoError(t, err)
+	}
+
+	err = pdb.CommitRoundAccounting(state.RoundUpdates, test.Round, &itypes.Block{})
+	assert.NoError(t, err, "failed to commit")
+
+	// Check that the asset is deleted.
+	{
+		row := db.QueryRow(
+			"SELECT deleted, closed_at FROM asset WHERE index = $1", int64(assetID))
+
+		var deleted sql.NullBool
+		var closedAt sql.NullInt64
+		err := row.Scan(&deleted, &closedAt)
+		assert.NoError(t, err)
+
+		assert.Equal(t, sql.NullBool{Valid: true, Bool: true}, deleted)
+		assert.Equal(t, sql.NullInt64{Valid: true, Int64: int64(test.Round)}, closedAt)
+	}
+
+	// Check that the account's asset holding is deleted.
+	{
+		row := db.QueryRow(
+			"SELECT assetid, deleted, closed_at FROM account_asset WHERE addr = $1",
+			test.AccountA[:])
+
+		var id int64
+		var deleted sql.NullBool
+		var closedAt sql.NullInt64
+		err := row.Scan(&id, &deleted, &closedAt)
+		assert.NoError(t, err)
+
+		assert.Equal(t, assetID, uint64(id))
+		assert.Equal(t, sql.NullBool{Valid: true, Bool: true}, deleted)
+		assert.Equal(t, sql.NullInt64{Valid: true, Int64: int64(test.Round)}, closedAt)
+	}
+}
+
+func TestDestroyAssetDeleteCreatorsHolding(t *testing.T) {
+	db, connStr, shutdownFunc := setupPostgres(t)
+	defer shutdownFunc()
+
+	pdb, err := idb.IndexerDbByName("postgres", connStr, nil, nil)
+	assert.NoError(t, err)
+
+	cache, err := pdb.GetDefaultFrozen()
+	assert.NoError(t, err)
+
+	assetID := uint64(3)
+
+	state := getAccounting(test.Round, cache)
+
+	// Create an asset.
+	{
+		// Create a transaction where all special addresses are different from creator's address.
+		txn := types.SignedTxnWithAD{
+			SignedTxn: types.SignedTxn{
+				Txn: types.Transaction{
+					Type: "acfg",
+					Header: types.Header{
+						Sender:     test.AccountA,
+					},
+					AssetConfigTxnFields: types.AssetConfigTxnFields{
+						AssetParams: types.AssetParams{
+							Manager:       test.AccountB,
+							Reserve:       test.AccountB,
+							Freeze:        test.AccountB,
+							Clawback:      test.AccountB,
```

Comment:
These are all AccountB so the test isn't working quite as designed according to the assertions.

Reply:
Why? We test that creator's asset holding is deleted and manager's is not.

Reply:
Ok that makes sense. I think I missed that AccountC opts-in below and you check for that as well.